flows/actions/send_email.go

```diff
@@ -109,11 +110,6 @@ func (a *SendEmailAction) Inspect(inspect func(flows.Inspectable)) {
 }
 
 // EnumerateTemplates enumerates all expressions on this object and its children
-func (a *SendEmailAction) EnumerateTemplates(include flows.TemplateIncluder) {
-	include.String(a.Subject)
-	include.String(a.Body)
-	include.Slice(a.Addresses)
-
-	include.Translations(a, "subject")
-	include.Translations(a, "body")
+func (a *SendEmailAction) EnumerateTemplates(localization flows.Localization, include func(string)) {
```

Comment:
Can we make this a mixin so we don't have all this repetition?

Reply:
Alas no because a golang "mixin" only knows about itself.. it can't point to the wider struct that contains it.

Right now the `Inspect()` stuff is going as deep as the actions, and then the new reflection stuff goes into the actions themselves. We need all this boilerplate because we need the reflection stuff to start on the actual struct itself and not just an embedded part of each action struct.

But if we got rid of the old inspection stuff all together we wouldn't need this because from the node, you can see the the actual action stucts...

Reply:
There must be some other way no? What if the mixin instance is constructed with the outer struct reference? Makes construction more complicated.

Reply:
You could make something like that work, but as you say it makes construction more complicated, and wouldn't it be good to use the same reflection-based system for the whole flow tree?

Reply:
Ok, maybe I'm not understanding that right. Just smells to have all that repetition and seems prone to implementations drifting in ways that break things. (there would be constructor repetition yes, but implementation could be done in one place that would affect all) Up to you.

Reply:
I'm not arguing against removing the repetition - I'm arguing for getting rid of it by making nodes use the same reflection-based inspection. That feels cleaner and saner.